addons/website_event_crm/models/event_lead_rule.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models
+
+
+class EventLeadRule(models.Model):
```

Comment:
I noticed that as you wrote, the language of visitors is the one they had when they landed on the website. So if I go to odoo . com in english, then change to French, the visitor is in english.

It is defined as so actually:
`    lang_id = fields.Many2one('res.lang', string='Language', help="Language from the website when visitor has been created")
`

I am not sure but this could be an important point to deal with. When registering, do we propagate the current language somewhere? On the registration for instance? We may use that instead. I am not sure though.

Reply:
On registration to the event? I don't think so as you might register a country but not a language (and countries can have several languages).

If a partner is linked to a lead then yes, their language will be transfered

I'll message luc to see what he thinks